code/modules/hydroponics/fermenting_barrel.dm

```diff
@@ -1,78 +1,54 @@
-	if(fruit.reagents)
-		fruit.reagents.trans_to(src, fruit.reagents.total_volume)
-	var/amount = fruit.seed.potency / 4
-	if(fruit.distill_reagent)
-		reagents.add_reagent(fruit.distill_reagent, amount)
-	else
-		var/data = list()
-		data["names"] = list("[initial(fruit.name)]" = 1)
-		data["color"] = fruit.filling_color
-		data["boozepwr"] = fruit.wine_power
-		if(fruit.wine_flavor)
-			data["tastes"] = list(fruit.wine_flavor = 1)
-		else
-			data["tastes"] = list(fruit.tastes[1] = 1)
-		reagents.add_reagent(/datum/reagent/consumable/ethanol/fruit_wine, amount, data)
-	qdel(fruit)
-	playsound(src, 'sound/effects/bubbles.ogg', 50, TRUE)
-
-/obj/structure/fermenting_barrel/attackby(obj/item/I, mob/user, params)
-	var/obj/item/reagent_containers/food/snacks/grown/fruit = I
-	if(istype(fruit))
-		if(!fruit.can_distill)
-			to_chat(user, "<span class='warning'>You can't distill this into anything...</span>")
-			return TRUE
-		else if(!user.transferItemToLoc(I,src))
-			to_chat(user, "<span class='warning'>[I] is stuck to your hand!</span>")
-			return TRUE
-		to_chat(user, "<span class='notice'>You place [I] into [src] to start the fermentation process.</span>")
-		addtimer(CALLBACK(src, .proc/makeWine, fruit), rand(80, 120) * speed_multiplier)
-		return TRUE
-	else
+/obj/structure/fermenting_barrel/attackby(obj/item/W, mob/user, params)
+	if(is_type_in_list(W, already_brewing))
+		to_chat(user, "You're already brewing that!")
+		return FALSE
+	var/brewing_result = W.on_brew() // list("reagents" = list("reagent1", "reagent2"), "booze_power" = 420, "prefix" = "arse")
+	if(istype(W, /obj/item/reagent_containers) && !brewing_result)
+		return FALSE
+	if(!brewing_result)
 		return ..()
+	if(user.temporarilyRemoveItemFromInventory(W))
+		seedify(W, rand(1,4))
+		brewables += list(brewing_result)
+		already_brewing += W.type
+		to_chat(user, "You insert [W] into [src].")
+		qdel(W)
 
 /obj/structure/fermenting_barrel/attack_hand(mob/user)
-	open = !open
-	if(open)
-		DISABLE_BITFIELD(reagents.flags, DRAINABLE)
-		ENABLE_BITFIELD(reagents.flags, REFILLABLE)
-		to_chat(user, "<span class='notice'>You open [src], letting you fill it.</span>")
-	else
-		ENABLE_BITFIELD(reagents.flags, DRAINABLE)
-		DISABLE_BITFIELD(reagents.flags, REFILLABLE)
-		to_chat(user, "<span class='notice'>You close [src], letting you draw from its tap.</span>")
-	update_icon()
-
-/obj/structure/fermenting_barrel/update_icon()
-	if(open)
-		icon_state = "barrel_open"
-	else
-		icon_state = "barrel"
+	if(brewables.len)
+		to_chat(user, "You brew a batch of ale.")
+		var/obj/item/reagent_containers/food/drinks/wooden_mug/AB = new(get_turf(src))
+		var/datum/reagent/consumable/ethanol/customizable/ale = AB.reagents.add_reagent(/datum/reagent/consumable/ethanol/customizable, 50)
+		var/current_highest_boozepwr = list("prefix" = "christaincode", "power" = 0)
+		for(var/A in brewables)
+			var/list/christaincode = A["reagents"]
+			for(var/R in christaincode)
```

Comment:
Could this be a more self-documenting variable name?

Reply:
its a temp holder var so idk what to name it

Reply:
Why not, like, "addreagents" or something?

Reply:
done